### ISSUE TITLE
[Clang][Frontend] Clang allows wrong usage of `ownership_returns`.

### ISSUE THREAD

**Issue by pskrgag** [llvm/llvm-project#99501]
Clang does not report an error for obviously wrong usage of `ownership_returns`  attribute. 

`void foo(void) __attribute__((ownership_returns(foo)));`

In above example, function returns nothing, so `ownership_returns` does not make any sense.

Example on godbolt (trunk): https://godbolt.org/z/xsGaE6e8r

**Comment by pskrgag** [comment]
Also weird thing

```
int foo(void) __attribute__((ownership_returns(foo)));
int foo1(int) __attribute__((ownership_takes(foo, 1)));
```

Clang reports `'ownership_takes' attribute only applies to pointer arguments` , but allows `int` as return value for `foo`.  

Could I get assigned for this?

**Comment by shafik** [comment]
It looks like this is static analyzer attribute: https://clang.llvm.org/docs/analyzer/checkers.html#experimental-checkers

CC @AaronBallman

**Comment by pskrgag** [comment]
Yeah. I found this example during #98941. I think, it would be nice to have consistency between `owenership_returns` and `ownership_takes`, since now it's not the case and it's hard to write down the semantics of these attributes
